input/script_utils.go

```diff
@@ -1232,6 +2184,107 @@ func CommitScriptToRemoteConfirmed(key *btcec.PublicKey) ([]byte, error) {
 	return builder.Script()
 }
 
+// NewRemoteCommitScriptTree constructs a new script tree for the remote party
+// to sweep their funds after a hard coded 1 block delay.
+func NewRemoteCommitScriptTree(remoteKey *btcec.PublicKey,
+) (*CommitScriptTree, error) {
+
+	// First, construct the remote party's tapscript they'll use to sweep their
+	// outputs.
+	builder := txscript.NewScriptBuilder()
+	builder.AddData(schnorr.SerializePubKey(remoteKey))
+	builder.AddOp(txscript.OP_CHECKSIG)
+	builder.AddOp(txscript.OP_CHECKSEQUENCEVERIFY)
+
+	remoteScript, err := builder.Script()
+	if err != nil {
+		return nil, err
+	}
+
+	// With this script constructed, we'll map that into a tapLeaf, then
+	// make a new tapscript root from that.
+	tapLeaf := txscript.NewBaseTapLeaf(remoteScript)
+	tapScriptTree := txscript.AssembleTaprootScriptTree(tapLeaf)
+	tapScriptRoot := tapScriptTree.RootNode.TapHash()
+
+	// Now that we have our root, we can arrive at the final output script
+	// by tweaking the internal key with this root.
+	toRemoteOutputKey := txscript.ComputeTaprootOutputKey(
+		&TaprootNUMsKey, tapScriptRoot[:],
+	)
+
+	return &CommitScriptTree{
+		TaprootKey:    toRemoteOutputKey,
+		SettleLeaf:    tapLeaf,
+		TapscriptTree: tapScriptTree,
+		TapscriptRoot: tapScriptRoot[:],
+	}, nil
+}
+
+// TaprootCommitScriptToRemote constructs a taproot witness program for the
+// output on the commitment transaction for the remote party. For the top level
+// key spend, we'll use a NUMs key to ensure that only the script path can be
+// taken. Using a set NUMs key here also means that recovery solutions can scan
+// the chain given knowledge of the public key for the remote party. We then
+// commit to a single tapscript leaf that holds the normal CSV 1 delay
+// script.
+//
+// Our single tapleaf will use the following script:
+//
+//	<remotepubkey> OP_CHECKSIG
+//	OP_CHECKSEQUENCEVERIFY
+//
+// The CSV clause is a bit subtle, but OP_CHECKSIG will return true if it
+// succeeds, which then enforces our 1 CSV. The true will remain on the stack,
+// causing the script to pass. If the CHECKSIG fails, then a 0 will remain on
+// the stack.
+func TaprootCommitScriptToRemote(remoteKey *btcec.PublicKey,
+) (*btcec.PublicKey, error) {
```

Comment:
Heh I think we need to stamp something in our style guide...I've been leaning towards how it is stylistically lately, but I know @guggero prefers the alternative. 

I think how the one as is lets the return value be on a single line.

Reply:
Yeah, my mental parser always throws a fit when seeing this :sweat_smile: I think mostly because the indentation isn't preserved, which is a whole new visual pattern to learn when scanning code...